super cool. probably a dumb question but what's the relationship between Riak and S3 here?

build fail real:
```
======================================================================
ERROR: setUpClass (corehq.blobs.tests.test_mixin.TestBlobMixinWithS3Backend)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/blobs/tests/test_mixin.py", line 353, in setUpClass
    raise SkipTest("S3_BLOB_DB_SETTINGS not configured")
SkipTest: S3_BLOB_DB_SETTINGS not configured
----------------------------------------------------------------------
```

--- a/corehq/blobs/s3db.py
+++ b/corehq/blobs/s3db.py
@@ -18,14 +18,14 @@ SAFENAME = re.compile("^[a-z0-9_./-]+$", re.IGNORECASE)
 
 class S3BlobDB(object):
 
-    def __init__(self, settings):
+    def __init__(self, config):
         self.db = boto3.resource(
             's3',
-            endpoint_url=settings.get("url"),
-            aws_access_key_id=settings.get("access_key"),
-            aws_secret_access_key=settings.get("secret_key"),
+            endpoint_url=config.get("url"),
+            aws_access_key_id=config.get("access_key"),
+            aws_secret_access_key=config.get("secret_key"),
         )
-        self.s3_bucket_name = settings.get("s3_bucket", DEFAULT_S3_BUCKET)
+        self.s3_bucket_name = config.get("s3_bucket", DEFAULT_S3_BUCKET)
         self.s3_bucket_exists = False
         # https://github.com/boto/boto3/issues/259
         self.db.meta.client.meta.events.unregister('before-sign.s3', fix_s3_host)

--- a/corehq/blobs/s3db.py
+++ b/corehq/blobs/s3db.py
@@ -1,6 +1,5 @@
 from __future__ import absolute_import
 import re
-from contextlib import contextmanager
 from uuid import uuid4
 
 from corehq.blobs import BlobInfo
@@ -54,12 +53,11 @@ class S3BlobDB(object):
             params = {"body": content, "headers": {}}
             calculate_md5(params)
             content_md5 = params["headers"]["Content-MD5"]
-        with self.s3_bucket(True) as s3_bucket:
-            obj = s3_bucket.put_object(
-                Key=path,
-                Body=content,
-                ContentMD5=content_md5,
-            )
+        obj = self.s3_bucket(create=True).put_object(
+            Key=path,
+            Body=content,
+            ContentMD5=content_md5,
+        )
         return BlobInfo(name, obj.content_length, "md5-" + content_md5)
 
     def get(self, name, bucket=DEFAULT_BUCKET):
@@ -73,14 +71,13 @@ class S3BlobDB(object):
         object should be closed when finished reading.
         """
         path = self.get_path(name, bucket)
-        with self.s3_bucket() as s3_bucket:
-            try:
-                resp = s3_bucket.Object(path).get()
-            except ClientError as err:
-                if err.response["Error"]["Code"] == "NoSuchKey":
-                    raise NotFound(name, bucket)
-                raise
-            return ClosingContextProxy(resp["Body"])  # body stream
+        try:
+            resp = self.s3_bucket().Object(path).get()
+        except ClientError as err:
+            if err.response["Error"]["Code"] == "NoSuchKey":
+                raise NotFound(name, bucket)
+            raise
+        return ClosingContextProxy(resp["Body"])  # body stream
 
     def delete(self, name=None, bucket=DEFAULT_BUCKET):
         """Delete a blob
@@ -93,21 +90,20 @@ class S3BlobDB(object):
         """
         path = self.get_path(name, bucket)
         success = True
-        with self.s3_bucket() as s3_bucket:
-            if name is None:
-                summaries = s3_bucket.objects.filter(Prefix=path + "/")
-                pages = ([{"Key": o.key} for o in page]
-                         for page in summaries.pages())
-            else:
-                pages = [[{"Key": path}]]
-            for objects in pages:
-                resp = s3_bucket.delete_objects(Delete={"Objects": objects})
-                if success:
-                    deleted = set(d["Key"] for d in resp.get("Deleted", []))
-                    success = all(o["Key"] in deleted for o in objects)
+        s3_bucket = self.s3_bucket()
+        if name is None:
+            summaries = s3_bucket.objects.filter(Prefix=path + "/")
+            pages = ([{"Key": o.key} for o in page]
+                     for page in summaries.pages())
+        else:
+            pages = [[{"Key": path}]]
+        for objects in pages:
+            resp = s3_bucket.delete_objects(Delete={"Objects": objects})
+            if success:
+                deleted = set(d["Key"] for d in resp.get("Deleted", []))
+                success = all(o["Key"] in deleted for o in objects)
         return success
 
-    @contextmanager
     def s3_bucket(self, create=False):
         if create and not self.s3_bucket_exists:
             try:
@@ -117,7 +113,7 @@ class S3BlobDB(object):
                     raise
                 self.db.create_bucket(Bucket=self.s3_bucket_name)
             self.s3_bucket_exists = True
-        yield self.db.Bucket(self.s3_bucket_name)
+        return self.db.Bucket(self.s3_bucket_name)
 
     @staticmethod
     def get_unique_name(basename):

--- a/corehq/blobs/tests/test_init.py
+++ b/corehq/blobs/tests/test_init.py
@@ -29,6 +29,6 @@ def test_get_blobdb(self, msg, root=True, blob_dir=None):
             temp_dir=None,
             blob_dir=blob_dir,
         )
-        with override_settings(SHARED_DRIVE_CONF=conf):
+        with override_settings(SHARED_DRIVE_CONF=conf, S3_BLOB_DB_SETTINGS=None):
             with assert_raises(mod.Error, msg=re.compile(msg)):
                 mod.get_blob_db()

--- a/corehq/blobs/tests/util.py
+++ b/corehq/blobs/tests/util.py
@@ -62,9 +62,9 @@ class TemporaryS3BlobDB(TemporaryBlobDBMixin, S3BlobDB):
         if not self.s3_bucket_exists:
             return
         assert self.s3_bucket_name.startswith("test-"), self.s3_bucket_name
-        with self.s3_bucket() as s3_bucket:
-            summaries = s3_bucket.objects.all()
-            for page in summaries.pages():
-                objects = [{"Key": o.key} for o in page]
-                s3_bucket.delete_objects(Delete={"Objects": objects})
-            s3_bucket.delete()
+        s3_bucket = self.s3_bucket()
+        summaries = s3_bucket.objects.all()
+        for page in summaries.pages():
+            objects = [{"Key": o.key} for o in page]
+            s3_bucket.delete_objects(Delete={"Objects": objects})
+        s3_bucket.delete()

